tangelo/linq/target/target_qiskit.py

```diff
@@ -55,24 +55,28 @@ def simulate_circuit(self, source_circuit: Circuit, return_statevector=False, in
                 and requested by the user (if not, set to None).
         """
 
-        translated_circuit = translate_c(source_circuit, "qiskit")
+        translated_circuit = translate_c(source_circuit, "qiskit",
+                output_options={"save_measurements": save_mid_circuit_meas})
 
         # If requested, set initial state
         if initial_statevector is not None:
             if self._noise_model:
                 raise ValueError("Cannot load an initial state if using a noise model, with Qiskit")
             else:
                 n_qubits = int(math.log2(len(initial_statevector)))
-                initial_state_circuit = self.qiskit.QuantumCircuit(n_qubits, n_qubits)
+                n_registers = source_circuit._gate_counts.get("MEASURE", 0) + source_circuit.width
+                initial_state_circuit = self.qiskit.QuantumCircuit(n_qubits, n_registers)
```

Comment:
if not `save_mid_circuit_meas`, we would not need registers beyond `source_circuit.width`.

Reply:
OK